spec/models/spree/payment_spec.rb

```diff
@@ -356,6 +356,22 @@
             payment.void_transaction!
           end
         end
+
+        context "if payment has any adjustment" do
+          let!(:order) { create(:order) }
+          let!(:payment_method) { create(:payment_method, calculator: ::Calculator::FlatRate.new(preferred_amount: 10)) }
+
+          it "should create another adjustment and revoke the previous one" do
+            payment = create(:payment, order: order, payment_method: payment_method)
+            expect(order.all_adjustments.payment_fee.eligible.length).to eq(1)
+
+            payment.void_transaction!
+            expect(order.all_adjustments.payment_fee.eligible.length).to eq(0)
```

Comment:
No need to change here but this is a good example for the `change` expectation:
```suggestion
            expect { payment.void_transaction! }.
              to change { order.all_adjustments.payment_fee.eligible.length }.by(-1)
```